ocs/ocs_twisted.py

```diff
@@ -1,29 +1,113 @@
 import threading
 from contextlib import contextmanager
 
+
 class TimeoutLock:
-    def __init__(self):
+    def __init__(self, default_timeout=0):
+        """
+        Locking mechanism to be used by OCS Agents.
+
+        Args:
+            default_timeout (float, optional):
+                Sets the default timeout value for acquire calls.
+                Defaults to 0.
+        """
         self.job = None
-        self._lock = threading.Lock()
+        self._active = threading.Lock()
+        self._next = threading.Lock()
+        self._default_timeout = default_timeout
+
+    def acquire(self, timeout=None, job=None):
```

Comment:
This is an API change... but acquire_timeout is not.  Is the claim that no one in socs uses "acquire" directly so this won't break anything?